File: 977-Squares-of-a-Sorted-Array.java
import static java.lang.Math.abs;
class Solution {
    public int[] sortedSquares(int[] nums) {
        int left =0;
        int right = nums.length -1;
        int size = nums.length;
        int[] res = new int[size];
        Arrays.fill(res,0);

        while(left <= right){
            //System.out.println(res[size]);
            if(Math.abs(nums[left]) >= Math.abs(nums[right])){
                size --;
                res[size] = nums[left] * nums[left];
                left ++;
            }
            else if(Math.abs(nums[left]) < Math.abs(nums[right])){
                size --;
                res[size] = nums[right] * nums[right];
                right --;
            }
        }
        return res;
    }
}

File: 977-Squares-of-a-Sorted-Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = [0]*len(nums)
        left , right = 0 , len(nums)-1
        for num in range(len(nums)-1,-1,-1):
            if abs(nums[left]) < abs(nums[right]):
                res[num] = nums[right] * nums[right]
                right -= 1
            else:
                res[num] = nums[left] * nums[left]
                left += 1
        return res